python/lsst/pipe/base/graph/graph.py

```diff
@@ -0,0 +1,779 @@
+        # edge from None to the task, but None is not a node we actually want
+        # so remove it here. This is a consequence of using add_edges from and
+        # is an implementation detail.
+        if None in self._taskGraph.nodes():
+            self._taskGraph.remove_node(None)
+
+        # Graph of quanta relations
+        self._connectedQuanta = nx.DiGraph()
+        # Likewise to the `_taskGraph` above
+        self._connectedQuanta.add_edges_from(_datasetDictToEdgeIterator(self._datasetRefDict))
+        # Attempting to create a graph that contains a quantum unconnected from
+        # any other node will create an edge from None to the task, but None is
+        # not a node we actually want so remove it here. This is a consequence
+        # of using add_edges from and is an implementation detail.
+        if None in self._connectedQuanta.nodes():
+            self._connectedQuanta.remove_node(None)
+
+    @property
+    def taskGraph(self) -> nx.DiGraph:
+        """Returns a graph representing the relations between the tasks inside
+        the quantum graph.
+
+        Returns
+        -------
+        taskGraph : `networkx.Digraph`
+            Internal datastructure that holds relations of `TaskDefs`s
+        """
+        return self._taskGraph
+
+    @property
+    def graph(self) -> nx.DiGraph:
+        """Returns a graph representing the relations between all the
+        `QuantumNode`s. Larely it should be preferred to iterate over, and use
+        methods of this class, but sometimes direct access to the networkx
+        object may be helpful
+
+        Returns
+        -------
+        graph : `networkx.Digraph`
+            Internal datastructure that holds relations of `QuantumNode`s
+        """
+        return self._connectedQuanta
+
+    def taskDefForNode(self, node: QuantumNode) -> TaskDef:
+        """Retrives the TaskDef associated with a `QuantumNode`.
+
+        Parameters
+        ----------
+        node : `QuantumNode`
+            Node for which the associated `TaskDef` is to be found
+
+        Returns
+        -------
+        `TaskDef`
+            The requested `TaskDef`
+        """
+        return self._quantumToTask[node.nodeNumber]
+
+    def getQuantumNodeByNodeNumber(self, nodeNumber: int) -> QuantumNode:
+        """Lookup a `QuantumNode` from a number associated with the node.
+
+        Parameters
+        ----------
+        nodeNumber : `int`
+            The number associated with a node
+
+        Returns
+        -------
+        `QuantumNode`
+            The node corresponding with input number
+
+        Raises
+        ------
+        `IndexError`
+            If the requested nodeNumber is greater than the number of nodes in
+            the graph.
+        """
+        return self._nodeList[nodeNumber]
+
+    @property
+    def inputQuanta(self) -> List[QuantumNode]:
+        """Makes a `list` of all `QuantumNode`s that are 'input' nodes to the
```

Comment:
Start-of-docstring verbs [should be in the imperative voice](https://developer.lsst.io/python/numpydoc.html#writing-summaries-for-functions-and-methods), i.e. "Make" rather than "Makes" here (also true for several methods below, with "Returns" -> "Return").